### ISSUE TITLE
Narrowing conversion from enum to float missed: fails to cause substitution failure

### ISSUE THREAD

**Issue by hubert-reinterpretcast** [llvm/llvm-project#21043]
|  |  |
| --- | --- |
| Bugzilla Link | [20669](https://llvm.org/bz20669) |
| Version | trunk |
| OS | All |
| CC | @DougGregor |

## Extended Description 
A narrowing conversion from an unscoped enumeration to a floating-point type does not trigger substitution failure in Clang.
The similar narrowing conversion from long long to float does.

ref: N3290 subclause 8.5.4 [dcl.init.list] paragraph 7.

### SOURCE:
```console
$ cat narrowingEnumToFloat.cc
```
```cpp
enum AA : long long { A = -1, B = 0x7fffffffffffffffLL } a;

template <typename T> void foo(T, char (*)[sizeof T{ a }] = 0);
int foo(int);

int main() {
   return foo(0.f);
}
```

### COMPILER INVOCATION AND OUTPUT:
```console
$ clang -cc1 -std=c++11 -x c++ narrowingEnumToFloat.cc 
narrowingEnumToFloat.cc:7:11: error: cannot initialize return object of type 'int' with an rvalue of type 'void'
   return foo(0.f);
          ^~~~~~~~
1 error generated.
```

### EXPECTED OUTPUT:
Successful compile.

### COMPILER VERSION INFO:
```console
$ clang++ -v
clang version 3.5.0 (trunk 214060)
Target: x86_64-unknown-linux-gnu
Thread model: posix
Found candidate GCC installation: /usr/lib/gcc/x86_64-linux-gnu/4.8
Found candidate GCC installation: /usr/lib/gcc/x86_64-linux-gnu/4.8.2
Found candidate GCC installation: /usr/lib/gcc/x86_64-linux-gnu/4.9
Found candidate GCC installation: /usr/lib/gcc/x86_64-linux-gnu/4.9.0
Selected GCC installation: /usr/lib/gcc/x86_64-linux-gnu/4.8
Candidate multilib: .;@m64
Selected multilib: .;@m64
```

**Comment by frederick-vs-ja** [comment]
Fixed in Clang 7 ([Godbolt link](https://godbolt.org/z/qfWzbxsTh)).
